public/app/features/dashboard/utils/panel.ts

```diff
@@ -39,7 +41,14 @@ export const removePanel = (dashboard: DashboardModel, panel: PanelModel, ask: b
     });
     return;
   }
+
   dashboard.removePanel(panel);
+  reduxStore.dispatch(
```

Comment:
Think best to only do this viewPanel !== null , not possible to delete from edit mode so there should be no need to nullify editPanel and tab

Reply:
It's possible to delete if you use the `p r` shortcut

Reply:
Interestingly, seems to not delete the dashboard your editing though :)  Should probably not allow delete in either view or edit mode

Reply:
So just disallow use of the shortcut in those modes? Any thoughts from a UX perspective @jessover9000 ?

Reply:
Hmm good question. I guess deleting during edit mode doesn't make that much sense, the usual UX pattern would be either deleting or editing and not both at the same time. But why should you not be able to delete in view mode?

Reply:
Did I understand the suggested behavior correct like this:

1. You should only be able to use the `p r` shortcut when in settings? Not in view or edit mode.
2. Not display or disable the "delete dashboard" button in settings if you currently is in edit mode.

I'm not sure I am a big fan of the solution in step 2. since the settings will behave differently depending on how you navigate to it. But hopefully I missed something.

Reply:
> You should only be able to use the p r shortcut when in settings? Not in view or edit mode.

in settings? no in the dashboard (non edit/view mode). 

> I'm not sure I am a big fan of the solution in step 2. since the settings will behave differently depending on how you navigate to it. But hopefully I missed something.

not sure where you get settings from :) what settings do you mean? 

All we are talking about is disabling the shortcut while in edit & view mode, and the delete menu option from the view mode (already removed from edit mode)

Reply:
Hehe, so then I am totally lost.

![lost](https://media.giphy.com/media/Bp3dFfoqpCKFyXuSzP/giphy.gif)

Need to re-test this one to give proper feedback.